File: CS/ClockTimeEditControl/ClockTimeEditViewInfo.cs
using System;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.ViewInfo;

namespace ClockTimeEditControl
{
    public class ClockTimeEditViewInfo : PictureEditViewInfo
    {
        public ClockTimeEditViewInfo(RepositoryItem item)
            : base(item)
        {
        }

        public override object EditValue
        {
            get { return base.EditValue; }
            set
            {
                if ( value is DateTime )
                    value = ((RepositoryItemClockTimeEdit)Item).ConvertTimeToClockImage((DateTime)value);

                base.EditValue = value;
            }
        }
    }
}

File: VB/ClockTimeEditControl/ClockTimeEditViewInfo.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.ViewInfo

Namespace ClockTimeEditControl
	Public Class ClockTimeEditViewInfo
		Inherits PictureEditViewInfo
		Public Sub New(ByVal item As RepositoryItem)
			MyBase.New(item)
		End Sub

		Public Overrides Property EditValue() As Object
			Get
				Return MyBase.EditValue
			End Get
			Set(ByVal value As Object)
				If TypeOf value Is DateTime Then
					value = (CType(Item, RepositoryItemClockTimeEdit)).ConvertTimeToClockImage(CDate(value))
				End If

				MyBase.EditValue = value
			End Set
		End Property
	End Class
End Namespace
